File: .codex/session_tracker.rb
#!/usr/bin/env ruby
# Session task logging for Codex CLI sessions
# Records tasks to .codex/SESSION_TASKS.md and provides lookup on restart.
require 'time'

module SessionTracker
  TASKS_FILE = File.expand_path('SESSION_TASKS.md', __dir__)

  # Append a new task description with UTC timestamp
  # description - String full task text
  def self.log_task(description)
    timestamp = Time.now.utc.iso8601
    File.open(TASKS_FILE, 'a') do |f|
      f.puts "#{timestamp}  #{description.strip}" 
    end
  rescue Errno::ENOENT
    Dir.mkdir(File.dirname(TASKS_FILE)) unless Dir.exist?(File.dirname(TASKS_FILE))
    retry
  end

  # Retrieve the last logged task entry, or nil if none
  # Returns the raw line from the tasks file
  def self.last_task
    return nil unless File.exist?(TASKS_FILE)
    File.readlines(TASKS_FILE, chomp: true).last
  end
end

File: .codex/session_tracker.js
#!/usr/bin/env node
// Session task logging for Codex CLI sessions (Node version)
// Records tasks to .codex/SESSION_TASKS.md and provides lookup on restart.
import fs from 'fs';
import path from 'path';

// Directory and tasks file path relative to working directory
const CODEX_DIR = path.resolve(process.cwd(), '.codex');
const TASKS_FILE = path.join(CODEX_DIR, 'SESSION_TASKS.md');

/**
 * Append a new task description with UTC timestamp
 * @param {string} description - full task text
 */
export function logTask(description) {
  try {
    fs.mkdirSync(CODEX_DIR, { recursive: true });
  } catch {
    // ignore
  }
  const timestamp = new Date().toISOString();
  const line = `${timestamp}  ${description.trim()}\n`;
  fs.appendFileSync(TASKS_FILE, line);
}

/**
 * Retrieve the last logged task entry, or null if none
 * @returns {string|null} raw line from the tasks file
 */
export function lastTask() {
  try {
    const data = fs.readFileSync(TASKS_FILE, 'utf8');
    const lines = data.split(/\r?\n/).filter(l => l.trim());
    return lines.length > 0 ? lines[lines.length - 1] : null;
  } catch {
    return null;
  }
}